mlflow/deployments/databricks/__init__.py

```diff
@@ -351,42 +352,67 @@ def create_endpoint(self, name, config=None, route_optimized=False):
-                name="chat",
                 config={
-                    "served_entities": [
-                        {
-                            "name": "test",
-                            "external_model": {
-                                "name": "gpt-4",
-                                "provider": "openai",
-                                "task": "llm/v1/chat",
-                                "openai_config": {
-                                    "openai_api_key": "{{secrets/scope/key}}",
+                    "name": "test",
+                    "config": {
+                        "served_entities": [
+                            {
+                                "external_model": {
+                                    "name": "gpt-4",
+                                    "provider": "openai",
+                                    "task": "llm/v1/chat",
+                                    "openai_config": {
+                                        "openai_api_key": "{{secrets/scope/key}}",
+                                    },
                                 },
-                            },
-                        }
-                    ],
+                            }
+                        ],
+                        "route_optimized": True,
+                    },
                 },
-                route_optimized=True,
             )
             assert endpoint == {
-                "name": "chat",
+                "name": "test",
                 "creator": "alice@company.com",
                 "creation_timestamp": 0,
                 "last_updated_timestamp": 0,
                 "state": {...},
                 "config": {...},
                 "tags": [...],
                 "id": "88fd3f75a0d24b0380ddc40484d7a31b",
+                "permission_level": "CAN_MANAGE",
+                "route_optimized": False,
+                "task": "llm/v1/chat",
+                "endpoint_type": "EXTERNAL_MODEL",
+                "creator_display_name": "Alice",
+                "creator_kind": "User",
             }
 
         """
-        config = config.copy() if config else {}  # avoid mutating config
-        extras = {}
-        for key in ("tags", "rate_limits"):
-            if tags := config.pop(key, None):
-                extras[key] = tags
-        payload = {"name": name, "config": config, "route_optimized": route_optimized, **extras}
+        if config and "config" in config:
+            # config contains the full API request payload
+            if route_optimized or name:
+                raise ValueError(
+                    "'route_optimized' and 'name' parameters cannot be used when "
+                    "'config' contains the full API request payload. Include "
+                    "'route_optimized' and/or 'name' in the payload instead."
+                )
+            payload = config
+        else:
+            # for backwards compatibility
+            warnings.warn(
```

Comment:
See above comment for how we should restructure the deprecation warning here (the messages will likely need to be appended to a list and separate messages for the individual fields will be required to be emitted in the warning message).